web/src/SidebarIcon.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { PureComponent } from "react"
+import { TriggerMode, RuntimeStatus } from "./types"
+import { Color } from "./constants"
+
+type SidebarIconProps = {
+  triggerMode: TriggerMode
+  status: RuntimeStatus
+  hasWarning: boolean
+  isBuilding: boolean
+}
+
+export enum IconType {
+  DotAuto = "dotAuto",
+  DotAutoPending = "dotAutoPending",
+  DotAutoBuilding = "dotAutoBuilding",
+  DotManual = "dotManual",
+  DotManualBuilding = "dotManualBuilding",
```

Comment:
doesn't have to be in this PR, but we should probably have a `ManualPending` option (e.g. you've clicked "build" for resource X but there are other things ahead of it in the queue so it hasn't started building yet)

Reply:
(except, wait, i don't mean "pending" as you use it below for "this resource hasn't come up yet at all"... i think there needs to be a way to indicate "waiting to build this" without scrubbing the color that shows you the current status of the resource.)